src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+
+  resize() {
+    // If we're using the default camera, update the aspect ratio
+    if (this.cameraType === 'default') {
+      this._computeCameraDefaultSettings();
+      this._setDefaultCamera();
+    } else {
+      this.perspective(
+        this.cameraFOV,
+        this.fbo.width / this.fbo.height
+      );
+    }
+  }
+}
+
+p5.FramebufferCamera = FramebufferCamera;
+
+class FramebufferTexture {
+  constructor(framebuffer, property) {
+    this.framebuffer = framebuffer;
+    this.property = property;
+  }
+
+  get width() {
+    return this.framebuffer.width * this.framebuffer.density;
+  }
+
+  get height() {
+    return this.framebuffer.height * this.framebuffer.density;
+  }
+
+  rawTexture() {
+    return this.framebuffer[this.property];
+  }
+}
+
+p5.FramebufferTexture = FramebufferTexture;
+
+class Framebuffer {
+  /**
+   * @class p5.Framebuffer
+   * @constructor
+   * @param {p5.Graphics|p5} target A p5 global instance or p5.Graphics
+   * @param {Object} [settings] A settings object
+   */
+  constructor(target, settings = {}) {
+    this.target = target;
+    this.target._renderer.framebuffers.add(this);
+
+    this.format = settings.format || constants.UNSIGNED_BYTE;
+    this.channels = settings.channels || (
+      target._renderer._pInst._glAttributes.alpha
+        ? constants.RGBA
+        : constants.RGB
+    );
+    this.depthFormat = settings.depthFormat || (
+      this.format === constants.UNSIGNED_BYTE
+        ? constants.UNSIGNED_BYTE
+        : constants.FLOAT
+    );
+    this.antialias = (
+      settings.antialias === undefined
+        ? target._pInst._glAttributes.antialias
+        : settings.antialias
+    );
+    if (this.antialias && target.webglVersion !== constants.WEBGL2) {
+      console.warn('Antialiasing is unsupported in a WebGL 1 context');
+      this.antialias = false;
+    }
+    if (settings.width && settings.height) {
+      this.width = settings.width;
+      this.height = settings.height;
+      this.autoSized = false;
+    } else {
+      this.width = target.width;
+      this.height = target.height;
+      this.autoSized = true;
+    }
+    this.density = settings.density || target.pixelDensity();
+
+    const gl = target._renderer.GL;
+    this.gl = gl;
+
+    this._checkIfFormatsAvailable();
+
+    this.framebuffer = gl.createFramebuffer();
+    if (!this.framebuffer) {
+      throw new Error('Unable to create a framebuffer');
+    }
+    if (this.antialias) {
+      this.aaFramebuffer = gl.createFramebuffer();
+      if (!this.aaFramebuffer) {
+        throw new Error('Unable to create a framebuffer');
+      }
+    }
+
+    this._recreateTextures();
+
+    const prevCam = this.target._renderer._curCamera;
+    this.defaultCamera = this.createCamera();
+    this.target._renderer._curCamera = prevCam;
+  }
+
+  resizeCanvas(width, height) {
+    this.autoSized = false;
+    this.width = width;
+    this.height = height;
+    this._handleResize();
+  }
+
+  pixelDensity(density) {
```

Comment:
For my own understanding...If I set the pixelDensity after the framebuffer has been constructed, it will then disable the auto resizing?

Reply:
Right. I wasn't sure, do you think it makes sense to have it part of width/height autosizing? In any case, I'll make sure this is mentioned in the docs.

Reply:
Hmm, I'm not sure I follow what you mean with width/height autoresizing. But I think as long as we document this behavior we should be ok.

Reply:
Right now autosizing is width+height+density, but potentially it could just be width+height that autosize and we let you change density independently